parts/linux/cloud-init/artifacts/ensure_imds_restriction.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+# Ensure IMDS restriction rule is correctly set on the node, this script will be registered as kubelet service's ExecStart
+# to ensure desired iptables rule will be ensured after the node is restarted.
+# Below 3 variables are passed from kubelet service's drop-in file.
+primaryNicIP="${PRIMARY_NIC_IP}"
+insertRuleToMangleTable="${INSERT_IMDS_RESTRICTION_RULE_TO_MANGLE_TABLE}"
+enableIMDSRestriction="${ENABLE_IMDS_RESTRICTION}"
+
+ensureIMDSRestrictionRule() {
+    if [[ $insertRuleToMangleTable == true ]]; then
+        echo "Before inserting IMDS restriction rule to mangle table, checking whether the rule already exists..."
+        iptables -t mangle -S | grep -- '-d 169.254.169.254/32 -p tcp -m tcp --dport 80 -m comment --comment "AKS managed: added by AgentBaker ensureIMDSRestriction for IMDS restriction feature" -j DROP'
+        if [[ $? -eq 0 ]]; then
+            echo "IMDS restriction rule already exists in mangle table, returning..."
+            return 0
+        fi
+        echo "Inserting IMDS restriction rule to mangle table..."
+        iptables -t mangle -I FORWARD 1 ! -s "$primaryNicIP" -d 169.254.169.254/32 -p tcp -m tcp --dport 80 -m comment --comment "AKS managed: added by AgentBaker ensureIMDSRestriction for IMDS restriction feature" -j DROP || exit ${ERR_INSERT_IMDS_RESTRICTION_RULE_INTO_MANGLE_TABLE}
```

Comment:
I don't believe you'll be able to access these exit codes anymore - you'd need to define them as constants within the script itself or simply `exit 1` instead

Reply:
updated the code to directly `exit 1`